src/Forms/ElementalAreaField.php

```diff
@@ -214,33 +217,85 @@ public function setSubmittedValue($value, $data = null)
         return $this->setValue(json_decode($value ?? '', true));
     }
 
+    /**
+     * This will perform FormField validation
+     * DataObject validation will happen in saveInto() as part of $element->write()
+     */
+    public function validate($validator)
+    {
+        $result = true;
+        foreach ($this->getElementData() as $arr) {
+            /** @var BaseElement $element */
+            /** @var array $data */
+            $element = $arr['element'];
+            $data = $arr['data'];
+            $elementForm = $this->getElementForm($element);
+            if (!$elementForm) {
+                continue;
+            }
+            $elementForm->loadDataFrom($data);
+            $formValidatorResult = $elementForm->validationResult();
```

Comment:
```suggestion
            $formValidatorResult = $elementForm->getValidator()->validate();
```
We should skip the logic in `validationResult()` which might skip validation - at this point we know for sure we want to validate the form, so we should just do it directly.

Reply:
Updated